packages/pigeon/lib/dart_generator.dart

```diff
@@ -252,15 +275,27 @@ void _writeFlutterApi(
                 indent.writeln('// ignore message');
                 call = 'api.${func.name}()';
               } else {
-                final String argType =
-                    _addGenericTypes(func.arguments[0].type, nullTag);
                 indent.writeln(
-                  'assert(message != null, \'Argument for $channelName was null. Expected $argType.\');',
+                  'assert(message != null, \'Argument for $channelName was null.\');',
                 );
+                const String argsArray = 'args';
                 indent.writeln(
-                  'final $argType input = (message as $argType$nullTag)$unwrapOperator;',
-                );
-                call = 'api.${func.name}(input)';
+                    'final List<Object$nullTag> $argsArray = (message as List<Object$nullTag>$nullTag)$unwrapOperator;');
+                final Iterable<String> argNames = indexMap(
+                    func.arguments, (int count, NamedType arg) => 'arg$count');
+                final Iterable<String> argTypes = func.arguments
+                    .map((NamedType e) => _addGenericTypes(e.type, nullTag));
+                enumerate(zip(argTypes, argNames),
+                    (int count, Tuple<String, String> element) {
+                  final String argType = element.first;
+                  final String argName = element.second;
+                  indent.writeln(
+                      'final $argType$nullTag $argName = $argsArray[$count] as $argType$nullTag;');
+                  indent.writeln(
+                      'assert($argName != null, \'Argument for $channelName was null. Expected $argType.\');');
```

Comment:
How about `Expected a value of type "argType"`. I found that the generated error messages read oddly as-is, since `null` is a value but this is a type.

Reply:
Reworded to "Argument for com.foo.bar was null, expected non-null int."

FWIW this shouldn't be able to be triggered by our users, this should only happen as a result of an error inside of the pigeon code (or someone abusing the generated code).